File: easy/longest_common_prefix.py
import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        sol = ""
        for i in range(len(strs[0])):
            check = False
            for str in strs[1:]:
                if i > len(str)-1 or strs[0][i] != str[i]:
                    check = True
                    break
            if check:
                return sol
            sol = sol + strs[0][i]
        return sol

File: easy/longest_common_prefix.js
var longestCommonPrefix = function (strs) {
  if (strs.length == 1) {
    return strs[0];
  }
  let str = "";
  for (let i = 0; i < strs[0].length + 1; i++) {
    str_ = strs[0].substring(0, i);
    let check = true;
    for (let j = 1; j < strs.length; j++) {
      if (str_ != strs[j].substring(0, i)) {
        check = false;
        break;
      }
    }
    if (!check) {
      return str;
    }
    str = str_;
  }
  return str;
};

// test
console.log(longestCommonPrefix(["flower", "flow", "flight"]));
console.log(longestCommonPrefix(["dog", "racecar", "car"]));
console.log(longestCommonPrefix(["a"]));
console.log(longestCommonPrefix(["flower", "flower", "flower", "flower"]));
